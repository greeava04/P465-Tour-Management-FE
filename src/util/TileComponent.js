import React from 'react';
import FavoriteBorderIcon from '@mui/icons-material/FavoriteBorder';
import FavoriteIcon from '@mui/icons-material/Favorite';
import AddIcon from "@mui/icons-material/Add"

const TileComponent = ({ location, onLocationSelect, onToggleFavorite, addToItinerary }) => {
    console.log(location)
    return (
        <div className="grid-item" onClick={() => onLocationSelect(location)}>
            <img src={location.pictureURL} alt="Location" />
            <div className="grid-text">
                <h3>{location.name}</h3>
                <p>Price: ${location.price}</p>
                <p>Rating: {location.rating}</p>
                <button onClick={(e) => { e.stopPropagation(); onToggleFavorite(location.name); }} className="grid-favorite-btn">
                    {location.isFavorited ? <FavoriteIcon style={{ color: '#F25C5C' }} /> : <FavoriteBorderIcon style={{ color: 'black' }} />}
                </button>
                <button onClick={(event) => addToItinerary(event, location)} className="grid-add-btn">
                    {location.isFavorited ? <AddIcon style={{ color: '#F25C5C' }} /> : <AddIcon style={{ color: 'black' }} />}
              </button>
            </div>
        </div>
    );
};

export default TileComponent;
