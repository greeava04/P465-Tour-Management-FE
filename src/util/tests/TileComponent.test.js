import renderer from 'react-test-renderer';
import TileComponent from '../TileComponent';
import EZTravelLogo from '../../images/EZTravelLogo.png'

it('renders page', () => {
    const location = {
        name: 'Sample Location 1', price: 150, rating: '★★★★☆', description: 'This is a sample description for Location 1.', pictureURL: EZTravelLogo, isFavorited: false
    }
    const component = renderer.create(
        <TileComponent location={location}></TileComponent>
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})