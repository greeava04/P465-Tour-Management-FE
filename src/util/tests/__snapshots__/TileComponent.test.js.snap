// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders page 1`] = `
<div
  className="grid-item"
  onClick={[Function]}
>
  <img
    alt="Location"
    src="EZTravelLogo.png"
  />
  <div
    className="grid-text"
  >
    <h3>
      Sample Location 1
    </h3>
    <p>
      Price: $
      150
    </p>
    <p>
      Rating: 
      ★★★★☆
    </p>
    <button
      className="grid-favorite-btn"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="FavoriteBorderIcon"
        focusable="false"
        style={
          Object {
            "color": "black",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3m-4.4 15.55-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05"
        />
      </svg>
    </button>
    <button
      className="grid-add-btn"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="AddIcon"
        focusable="false"
        style={
          Object {
            "color": "black",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z"
        />
      </svg>
    </button>
  </div>
</div>
`;
